### PR TITLE
add docker build and publish

### PR DESCRIPTION
**Pull request by lostsnow**
Docker image will publish to `ghcr.io/fluffos/fluffos`

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches: [ master ]
+    paths-ignore:
+      - 'docs/**'
+  pull_request:
```
Comment (thefallentree) [inline .github/workflows/docker-publish.yml:13]: we probably should not run on pull_request,

but we could run on new releases

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v1.10.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
Comment (thefallentree) [inline .github/workflows/docker-publish.yml:45]: what is the TOKEN here?

Comment (lostsnow) [reply .github/workflows/docker-publish.yml:45]: github auto generated: https://docs.github.com/en/actions/reference/authentication-in-a-workflow

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (lostsnow) [reply .github/workflows/docker-publish.yml]: Already changed, but we can't find the docker build failure when pull request

**Review by thefallentree — APPROVED** [review]

**Pull request merged**